src/main/java/org/jenkinsci/plugins/deployment/conditions/ThresholdCondition.java

```diff
@@ -36,6 +36,38 @@ public int getThreshold() {
         return threshold;
     }
 
+
+    @Override
+    public RangeSet calcMatchingBuildNumberOf(Job upstream, DeploymentFacet<?> facet, Job workflowJob) {
```

Comment:
Why do you need two `Job` parameters? It does not matter if it's a workflow job or not.